tests/test_builder.py

```diff
@@ -0,0 +1,335 @@
+from mandible.metadata_mapper.builder import (
+    add,
+    build,
+    floordiv,
+    mapped,
+    mul,
+    reformatted,
+    sub,
+    truediv,
+)
+
+
+def test_build_noop():
+    template = {
+        "something": ["simple"],
+        "with": {
+            "nested": [
+                {"data": ["structures"]},
+            ],
+        },
+    }
+
+    assert build(template) == template
+
+
+def test_build_mapped():
+    template = mapped(
+        source="some_source",
+        key="some.key",
+    )
+
+    assert build(template) == {
+        "@mapped": {
+            "source": "some_source",
+            "key": "some.key",
+        },
+    }
+
+
+def test_build_directive_marker():
+    template = mapped(
+        source="some_source",
+        key="some.key",
+    )
+
+    assert build(template, directive_marker="#%^") == {
+        "#%^mapped": {
+            "source": "some_source",
+            "key": "some.key",
+        },
+    }
+
+
+def test_build_reformatted():
+    template = reformatted(
+        format="Json",
+        value=mapped(
+            source="some_source",
+            key="some.key",
+        ),
+        key="foo",
+    )
+
+    assert build(template) == {
+        "@reformatted": {
+            "format": "Json",
+            "value": {
+                "@mapped": {
+                    "source": "some_source",
+                    "key": "some.key",
+                },
+            },
+            "key": "foo",
+        },
+    }
+
+
+def test_build_add():
+    template = add(
+        left=1,
+        right=2,
+    )
+
+    assert build(template) == {
+        "@add": {
+            "left": 1,
+            "right": 2,
+        },
+    }
+
+
+def test_build_add_automatic():
+    template = mapped(
+        source="some_source",
+        key="some.key",
+    ) + 10
+
+    assert build(template) == {
+        "@add": {
+            "left": {
+                "@mapped": {
+                    "source": "some_source",
+                    "key": "some.key",
+                }
```

Comment:
This file is still missing some trailing commas.

Reply:
Ah yep, they got lost during a rebase.